### DISCUSSION TITLE
Error in magento2 Recipe since v7.4.1

### DISCUSSION THREAD

**Discussion by Eikzx** [deployphp/deployer#3920] [Help needed]
Hello everyone,

since updating to v7.4.1 we have an issue deploying Magento 2 projects.

During the process of upgrading the DB we get following error:

```
magento:upgrade:db
00:10
[stage]  error  in magento2.php on line 303:
[stage] run /usr/bin/php ~/releases/2024-10-14_12-16-51/./bin/magento setup:db-schema:upgrade --no-interaction
[stage] err In ErrorHandler.php line 62:
[stage] err Warning: Trying to access array offset on value of type null in /var/www/re
[stage] err leases/2024-10-14_12-16-51/setup/src/Magento/Setup/Model/Installer.php on l
[stage] err ine 1049
```

We are running a gitlab-Runner.
The target is Ubuntu 22.04.

Going back to v7.4.0 solved the issue for us.

Kindest Regards
Eike

**Comment by antonmedv** [comment]
@peterjaap please take a look.

**Comment by carlos-reynosa** [comment]
I've seen this error before and it's probably a problem with your code. 

Last time I got this was because I had a module in the config.pho file that wasn't installed.

**Comment by Eikzx** [comment]
@carlos-reynosa I have checked my config.php and all modules listed there are also installed.

It also works with the version 7.4.0 flawlessly only 7.4.1 brings up this issue.
I will go ahead and double check it, just fr my sanity :D

**Comment by Aquive** [comment]
When I look at the release notes of v7.4.1 (https://github.com/deployphp/deployer/releases/tag/v7.4.1) I see some changes regarding, Magento 2.

- https://github.com/deployphp/deployer/pull/3812
- https://github.com/deployphp/deployer/pull/3818

The first one seems related? Maybe those will give a clue?

**Comment by peterjaap** [comment]
@Eikzx in 2.4.7-p3, that line is as follows; `$configVer = $this->moduleList->getOne($moduleName)['setup_version'];`

If that is the same line for you, it looks like a module is causing some issues. Set an xdebug breakpoint on that line and check the `$moduleName` variable contents to see which module is causing it.
